File: app/src/main/java/com/stupidtree/hichat/data/TypeConverters.java
package com.stupidtree.hichat.data;


import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.JsonElement;

import java.sql.Date;
import java.sql.Timestamp;

/**
 * ROOM需要使用转换器将时间戳转换为Date
 */
public class TypeConverters {
    @TypeConverter
    public static Long fromTimestamp(Timestamp value) {
        return value == null ? null : value.getTime();
    }

    @TypeConverter
    public static Timestamp dateToTimestamp(Long date) {
        return date == null ? null : new Timestamp(date);
    }

}

File: app/src/main/java/com/stupidtree/cloudliter/data/TypeConverters.kt
package com.stupidtree.cloudliter.data

import androidx.room.TypeConverter
import com.stupidtree.cloudliter.data.model.UserLocal
import java.sql.Timestamp

/**
 * ROOM需要使用转换器将时间戳转换为Date
 */
object TypeConverters {
    @JvmStatic
    @TypeConverter
    fun fromTimestamp(value: Timestamp?): Long? {
        return value?.time
    }

    @JvmStatic
    @TypeConverter
    fun dateToTimestamp(date: Long?): Timestamp? {
        return date?.let { Timestamp(it) }
    }

    @JvmStatic
    @TypeConverter
    fun genderToString(date: UserLocal.GENDER): String {
        return date.name
    }

    @JvmStatic
    @TypeConverter
    fun stringToGender(str:String): UserLocal.GENDER {
        return UserLocal.GENDER.valueOf(str)
    }



    @JvmStatic
    @TypeConverter
    fun typePermissionToString(date: UserLocal.TYPEPERMISSION): String {
        return date.name
    }

    @JvmStatic
    @TypeConverter
    fun stringToTypePermission(str:String): UserLocal.TYPEPERMISSION {
        return UserLocal.TYPEPERMISSION.valueOf(str)
    }
}